keras_cv/layers/preprocessing/rgb_shift_test.py

```diff
@@ -0,0 +1,84 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import tensorflow as tf
+
+from keras_cv.layers.preprocessing.rgb_shift import RGBShift
+
+
+class RGBShiftTest(tf.test.TestCase):
+    def test_return_shapes(self):
+        xs = tf.ones((2, 512, 512, 3))
+
+        layer = RGBShift(factor=1.0)
+
+        xs = layer(xs, training=True)
+        self.assertEqual(xs.shape, [2, 512, 512, 3])
+
+    def test_non_square_image(self):
+        xs = tf.cast(
+            tf.stack(
+                [2 * tf.ones((1024, 512, 3)), tf.ones((1024, 512, 3))],
+                axis=0,
+            ),
+            dtype=tf.float32,
+        )
+
+        layer = RGBShift(factor=[0.1, 0.3])
+
+        xs = layer(xs, training=True)
+        self.assertFalse(tf.math.reduce_any(xs[0] == 2.0))
```

Comment:
Is this consistently passing?  If you're using reduce_any xs[0] and it jitters all 3 channels wont this fail?

Reply:
Can you describe it a little more? I did run pytest, all ok though. It should not cause any issue:

```python

xs = 2 * tf.ones((1024, 1024, 3)) 

layer = RGBShift( .... )

xs = layer(xs) 

self.assertFalse(tf.math.reduce_any(xs == 2.0))

```